### DISCUSSION TITLE
How can I send notification to all using rest API without topic or token ?

### DISCUSSION THREAD

**Discussion by pk936** [invertase/react-native-firebase#4588] [Q&A] [answered]
I used 

`https://fcm.googleapis.com/v1/projects/proj-id/messages:send`

with request:

```
{
  "message": {
    "topic": "news",
    "notification": {
      "title": "Breaking News",
      "body": "New news story available."
    },
    "data": {
      "story_id": "story_12345"
    }
  }
}
```

Is adding a topic or token must in request ? They should be required only when I need to send notification to a group of devices. But I want to send to all devices.

**Comment by andersonaddo** [comment, marked as answer]
I guess one quick and painless option is to get every device to automatically subscribe to a topic just for these types of notifications.

**Reply by pk936** [reply]
ok.. Got it.

But this should be mentioned in docs that topic/token is mandatory. 

A newbie will obviously think that topic/token is for selected devices.
Subscribing every device to a topic (say "all"), does not make any sense. Instead topic/ token should be optional.

@andersonaddo  
And how **Notification Composer** in Firebase Console works then ? I did not mentioned any topic/token in its textbox and can get notificaion to all **testing devices**. Seems like **Notification Composer** does not need topic/token, is it true ?

**Reply by andersonaddo** [reply]
It's mentioned in the firebase docs, actually. I suggest you take a deep dive at those. 

Yeah I don't know how the firebase composer does what it does, tbh. Maybe it uses some secret topics or something.
